analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+      copykat_classification == "diploid" ~ "Normal",
+      TRUE ~ copykat_classification
+    )
+  )
+
+# infercnv output 
+infercnv_results_df <- readr::read_tsv(infercnv_predictions_file)
+infercnv_metadata_df <- read.delim(infercnv_metadata_file) |> 
+  tibble::rownames_to_column("barcodes")
+
+# remove MT and GL chr, comments in docs recommend that these are not useful and can be inaccurate 
+chr_columns_to_keep <- which(!stringr::str_detect(names(infercnv_metadata_df), "chrMT|chrGL"))
+infercnv_metadata_df <- infercnv_metadata_df[, chr_columns_to_keep]
+
+# gene set scores 
+geneset_scores_df <- readr::read_tsv(geneset_scores_file)
+```
+
+```{r}
+# grab all coldata and add in all classifications and results to use for plotting throughout notebook 
+# get list of marker genes to use 
+marker_genes <- marker_genes_df |> 
+  dplyr::filter(cell_type == "tumor") |> 
+  dplyr::pull(ensembl_gene_id)
+
+# get the gene expression counts for all marker genes
+marker_gene_exp <- logcounts(sce[marker_genes, ]) |>
+  as.matrix() |>
+  t() |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcodes")
+
+# pull out the UMAP coordinates and make a data frame to use for plotting
+coldata_df <- sce |> 
+  scuttle::makePerCellDF(use.dimred = "UMAP") |> 
+  # replace UMAP.1 with UMAP1
+  dplyr::rename_with(
+        \(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")
+      ) |> 
+  # add in marker gene expression to dataframe 
+  dplyr::left_join(marker_gene_exp, by = "barcodes") |> 
+  # combine all genes into a single column for easy faceting
+  tidyr::pivot_longer(
+    cols = starts_with("ENSG"),
+    names_to = "ensembl_gene_id",
+    values_to = "gene_expression"
+  ) |> 
+  # join with marker gene df to get gene symbols for plotting 
+  dplyr::left_join(marker_genes_df, by = c("ensembl_gene_id")) |>
+  # get rid of excess columns
+  dplyr::select(barcodes, UMAP1, UMAP2, gene_symbol, ensembl_gene_id, gene_expression) |> 
+  # add in marker gene classifications
+  dplyr::left_join(marker_gene_results_df) |> 
+  # add in copykat results 
+  dplyr::left_join(copykat_results_df) |> 
+  # add in infercnv results and metadata
+  dplyr::left_join(infercnv_results_df) |> 
+  dplyr::left_join(infercnv_metadata_df) |> 
+  # gene set scores 
+  dplyr::left_join(geneset_scores_df)
+
+```
+
+## Identify tumor cell population 
+
+Below is a UMAP showing which cells are labeled as tumor cells using each of the various classification methods. 
+
+```{r}
+# plot tumor cells for each classification method 
+tumor_umap_df <- coldata_df |> 
+  dplyr::select(barcodes, UMAP1, UMAP2, marker_gene_classification, copykat_classification, cnv_proportion_classification) |> 
+  tidyr::pivot_longer(
+    cols = ends_with("classification"),
+    names_to = "method",
+    values_to = "classification"
+  )
+
+ggplot(tumor_umap_df, aes(x = UMAP1, UMAP2, color = classification)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  facet_wrap(vars(method))
```

Comment:
I noted in this plot that the copykat classification does not have a reference class. Is this from the no-reference data?